cmd/argocd/commands/root.go

```diff
@@ -59,5 +59,6 @@ func NewCommand() *cobra.Command {
 	command.PersistentFlags().StringVar(&clientOpts.AuthToken, "auth-token", config.GetFlag("auth-token", ""), "Authentication token")
 	command.PersistentFlags().BoolVar(&clientOpts.GRPCWeb, "grpc-web", config.GetBoolFlag("grpc-web"), "Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2.")
 	command.PersistentFlags().StringVar(&logLevel, "loglevel", config.GetFlag("loglevel", "info"), "Set the logging level. One of: debug|info|warn|error")
+	command.PersistentFlags().StringSliceVar(&clientOpts.AddHeader, "add-header", []string{}, "Sets additional headers to all requests made by Argo CD CLI. (Can be repeated multiple times to add multiple headers, also supports comma separated headers)")
```

Comment:
We try and make our commands similar to `kubectl`. However, I can't see anything similar to that. `curl` has: `-H, --header <header>`. Perhaps we should be consistent with that?

Reply:
alright, changed cli flag to --header